### PR TITLE
Add 'play again' button for local/AI games

### PR DESCRIPTION
**Pull request by jasonharrison**
Add 'play again' button for local/AI games
---
image

### CODE REVIEW

File: src/App/Game/GameOver.tsx [hunk 1]
```diff
@@ -4,23 +4,47 @@ import { IGameArgs } from './GameBoardWrapper';
...
+      playAgainLink = (
+        <Button
+          onClick={this._refreshPage}
+          variant="outlined"
+          style={{ width: '150px', marginRight: '50%', marginLeft: '35%', marginTop: '8px' }}
+        >
+          <ReplayIcon style={{ marginRight: '8px' }} />
+          Play Again
+        </Button>);
+      playAgainText = 'Check out our games below:';
```
Comment (flamecoals) [inline src/App/Game/GameOver.tsx]: I don't like that this sentence ends with :, it feels weird. Besides that, LGTM

File: src/App/Game/GameOver.tsx [hunk 2]
```diff
@@ -4,23 +4,47 @@ import { IGameArgs } from './GameBoardWrapper';
...
+    } else {
+      playAgainText = 'Do you want to play again? Check out our games below:';
```
Comment (flamecoals) [inline src/App/Game/GameOver.tsx]: Here to,keep the period instead of :

File: src/games/chess/board.tsx [hunk 3]
```diff
@@ -51,7 +51,11 @@ export class Board extends React.Component<IBoardProps, {}> {
     }
     if (this.props.ctx.gameover) {
       return (
-        <GameLayout gameOver={this._getGameOver()} />
+        <GameLayout
+          gameOver={this._getGameOver()}
+          playAgain={((this.props.gameArgs) && ((this.props.gameArgs.mode === GameMode.AI)
```
Comment (flamecoals) [inline src/games/chess/board.tsx]: I don't think this logic should be here. Just send the gameArgs.mode to GameLayout and add this logic there instead.

File: src/games/tictactoe/board.tsx [hunk 4]
```diff
@@ -91,7 +95,7 @@ export class Board extends React.Component<IBoardProps, {}> {
   render() {
     if (this.props.ctx.gameover) {
       return (
-        <GameLayout gameOver={this._getGameOver()} />
+        <GameLayout gameOver={this._getGameOver()} playAgain={this.isAIGame() || this.isOfflineGame()} />
```
Comment (flamecoals) [inline src/games/tictactoe/board.tsx]: Same, I don't like to have to repeat the same logic at each game, better to just pass the game type so there is no duplication of code

File: src/App/Game/GameLayout.tsx [hunk 5]
```diff
@@ -6,12 +6,13 @@ import Typography from '@material-ui/core/Typography';
...
   children?: React.ReactNode;
   gameOver?: string;
+  playAgain?: boolean;
 }
 
 export class GameLayout extends React.Component<IGameLayoutProps, {}> {
   render() {
     if (this.props.gameOver) {
-      return (<GameOver result={this.props.gameOver} />);
+      return (<GameOver result={this.props.gameOver} playAgain={this.props.playAgain} />);
```
Comment (flamecoals) [inline src/App/Game/GameLayout.tsx]: Booleans are always bad parameters, it is better to pass gameArgs or the game mode

File: src/games/chess/board.tsx (hunk 3 above)
Comment (flamecoals) [reply src/games/chess/board.tsx]: The logic should be in Game over tbh

File: src/App/Game/GameLayout.tsx (hunk 5 above)
Comment (jasonharrison) [reply src/App/Game/GameLayout.tsx]: Agreed, changed

File: src/games/tictactoe/board.tsx [hunk 6]
```diff
@@ -47,6 +47,10 @@ export class Board extends React.Component<IBoardProps, {}> {
     return (this.props.gameArgs && this.props.gameArgs.mode === GameMode.OnlineFriend);
   }
 
+  isOfflineGame() {
```
Comment (flamecoals) [inline src/games/tictactoe/board.tsx]: This check is happening in GameOver.tsx now, you dont need this. This is also the reason the coverage is decreasing. After you delete this line it should be good to merge

Comment (jasonharrison) [reply src/games/tictactoe/board.tsx]: Thank you!

**Pull request merged**